### PR TITLE
File Name Validator Improvements

### PR DESCRIPTION
**Pull request by alperozturk96**
- [x] Tests written, or not not needed

### CODE REVIEW

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -1237,11 +1237,12 @@
...
     <string name="file_name_validator_rename_before_move_or_copy">%s. Please rename the file before moving or copying</string>
-    <string name="file_name_validator_upload_content_error">Some contents cannot able to uploaded due to contains reserved names or invalid character</string>
+    <string name="file_name_validator_upload_content_error">Some content cannot be uploaded since it contains reserved names or invalid characters</string>
     <string name="file_name_validator_error_contains_reserved_names_or_invalid_characters">Folder path contains reserved names or invalid character</string>
     <string name="file_name_validator_error_invalid_character">Name contains an invalid character: %s</string>
     <string name="file_name_validator_error_reserved_names">%s is a forbidden name</string>
     <string name="file_name_validator_error_forbidden_file_extensions">%s is a forbidden file extension</string>
+    <string name="file_name_validator_error_forbidden_space_character_extensions"> A space character is not allowed in file names</string>
```
Comment (provokateurin) [inline app/src/main/res/values/strings.xml:1245]:
```suggestion
    <string name="file_name_validator_error_forbidden_space_character_extensions">A space character is not allowed in file names</string>
```
That is ironic :P

**Review by ZetaTom — DISMISSED** [review]

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -1237,11 +1237,12 @@
     <string name="unified_search_fragment_permission_needed">Permissions are required to open search result otherwise it will redirected to web…</string>
     <string name="file_name_validator_current_path_is_invalid">Current folder name is invalid, please rename the folder. Redirecting to root</string>
     <string name="file_name_validator_rename_before_move_or_copy">%s. Please rename the file before moving or copying</string>
-    <string name="file_name_validator_upload_content_error">Some contents cannot able to uploaded due to contains reserved names or invalid character</string>
+    <string name="file_name_validator_upload_content_error">Some content cannot be uploaded since it contains reserved names or invalid characters</string>
```
[the remark this answers is not in the document]
Comment (alperozturk96) [reply app/src/main/res/values/strings.xml:1240]: @tobiasKaminsky suggested the first version and community accepted

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -1237,11 +1237,12 @@
...
     <string name="file_name_validator_error_contains_reserved_names_or_invalid_characters">Folder path contains reserved names or invalid character</string>
     <string name="file_name_validator_error_invalid_character">Name contains an invalid character: %s</string>
     <string name="file_name_validator_error_reserved_names">%s is a forbidden name</string>
     <string name="file_name_validator_error_forbidden_file_extensions">%s is a forbidden file extension</string>
+    <string name="file_name_validator_error_forbidden_space_character_extensions">A space character is not allowed in file names</string>
```
[the remark this answers is not in the document]
Comment (alperozturk96) [reply app/src/main/res/values/strings.xml:1245]: The community has already approved the first version

**Comment by ZetaTom** [comment]
@alperozturk96, first of all, my review is not invalidated because someone else has approved something. In fact, it is still *valid*, as the concerns raised in my comments still hold true.

To prove my point, I'll elaborate: as the string is set with this pull request, it suggests that no spaces shall be contained in any filename. This is a valid interpretation of *A space character is not allowed in file names*.

Obviously this isn't true, as you can still have a file called *This Filename Contains Spaces.md*. Because this is a perfectly valid filename, I would suggest rephrasing this sentence to reflect the real problem, the *trailing* spaces. So I suggested the following phrase instead: *Filenames must not contain any trailing spaces*, as this accurately reflects the problem and tells the user how to fix it.

So I ask you to please, simply, reconsider my review.

**Comment by alperozturk96** [comment]
@ZetaTom I think your suggestion touches on another detail and is not quite right. Because "Filenames must not contain any _trailing or leading spaces_" not just trailing.

Feedback was received from the “Mobile apps public” (Transifex) channel regarding the mentioned translations, and their current versions were accepted.

I have no problem with either the translation you suggested or the previously approved text. Both are suitable for me. There is an approved translation in place, and we need to consult the translators again based on your suggestion.

I don’t find it appropriate to portray even a single translation as a major issue. I cannot have back-and-forth exchanges between you and the translators. Such attitudes only slow us down and contribute negatively rather than positively. I don’t think engaging in lengthy arguments here will be beneficial.

As previously mentioned, I don’t believe the initial suggestion would be a massive problem for users. @rakekniven  @tobiasKaminsky

**Pull request merged**